File: FCFS.diffTime.py
class Process:
    def __init__(self, pid, arrival_time, burst_time):
        self.id = pid
        self.at = arrival_time
        self.bt = burst_time
        self.ct = 0
        self.tat = 0
        self.wt = 0

def input_processes(n):
    processes = []
    for i in range(n):
        arrival_time = int(input(f"Enter arrival time for process {i + 1}: "))
        burst_time = int(input(f"Enter burst time for process {i + 1}: "))
        processes.append(Process(i + 1, arrival_time, burst_time))
    return processes

def sort_processes(processes):
    return sorted(processes, key=lambda x: x.at)

def calculate_metrics(processes):
    sum_wt = 0
    sum_tat = 0
    for p in processes:
        p.wt = sum_wt
        sum_wt += p.bt
        p.ct = p.wt + p.bt
        p.tat = p.ct - p.at
        sum_tat += p.tat
    return sum_wt, sum_tat

def display_results(processes, total_wt, total_tat):
    print("\nProcess \tArrival\tBurst\tCompletion\tTurn Around\tWaiting")
    for p in processes:
        print(f"{p.id}\t\t{p.at}\t{p.bt}\t{p.ct}\t\t{p.tat}\t\t{p.wt}")
    print(f"\nTotal waiting time: {total_wt}")
    print(f"Total Turn-around time: {total_tat}")
    n = len(processes)
    print(f"Average waiting time: {total_wt / n:.2f}")
    print(f"Average Turn-around time: {total_tat / n:.2f}")

if __name__ == "__main__":
    print("Name: Amitkumar Vaghela, ERP No: 2203031050777")
    n = int(input("Enter the number of processes in your system: "))
    processes = input_processes(n)
    sorted_processes = sort_processes(processes)
    total_wt, total_tat = calculate_metrics(sorted_processes)
    display_results(sorted_processes, total_wt, total_tat)

File: FCFS_diffTime.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Process {
    int id;
    int at;
    int bt;
    int ct;
    int tat;
    int wt;
} Process;

void input(Process *p, int n) {
    for (int i = 0; i < n; i++) {
        printf("Enter arrival time for process %d: ", i + 1);
        scanf("%d", &p[i].at);
        printf("Enter burst time for process %d: ", i + 1);
        scanf("%d", &p[i].bt);
        p[i].id = i + 1;
    }
}

void sort(Process *p, int n) {
    for (int i = 0; i < n - 1; i++) {
        for (int j = 0; j < n - i - 1; j++) {
            if (p[j].at > p[j + 1].at) {
                Process temp = p[j];
                p[j] = p[j + 1];
                p[j + 1] = temp;
            }
        }
    }
}

void calc(Process *p, int n) {
    int sum = 0;
    for (int i = 0; i < n; i++) {
        sum += p[i].at;
        p[i].wt = sum;
        p[i].ct = p[i].wt + p[i].bt;
        p[i].tat = p[i].ct - p[i].at;
    }
}

void show(Process *p, int n) {
    int total_wt = 0, total_tat = 0;
    printf("\nProcess\tArrival\tBurst\tCompletion\tTurn Around\tWaiting\n");
    for (int i = 0; i < n; i++) {
        total_tat += p[i].tat;
        total_wt += p[i].wt;
        printf("%d\t\t%d\t%d\t%d\t\t%d\t%d\n", p[i].id, p[i].at, p[i].bt, p[i].ct, p[i].tat, p[i].wt);
    }
    printf("\nTotal waiting time: %d\n", total_wt);
    printf("Total Turn-around time: %d\n", total_tat);
    printf("Average waiting time: %.2f\n", (float)total_wt / n);
    printf("Average Turn-around time: %.2f\n", (float)total_tat / n);
}

int main() {
    int n;
    printf("Name: Vidhi Pateliya, ERP No: 2203031241412");
    printf("\nEnter the no of processes in your system: ");
    scanf("%d", &n);
    Process *p = malloc(n * sizeof(Process));
    input(p, n);
    sort(p, n);
    calc(p, n);
    show(p, n);
    free(p);
    return 0;
}